pkg/managed/clusterroles_test.go

```diff
@@ -0,0 +1,204 @@
+		},
+	}
+	_, err := clusterConnection.KubeClient.RbacV1().ClusterRoles().Create(context.TODO(), &cr, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("can't create cluster role: %v", err)
+	}
+
+	CleanupOrphanedClusterRoles(modelBindingPair, clusterConnections)
+
+	_, err = clusterConnection.KubeClient.RbacV1().ClusterRoles().Get(context.TODO(), "orphaned", metav1.GetOptions{})
+	if err == nil {
+		t.Fatal("expected cluster role to be cleaned up")
+	}
+}
+
+func TestCreateClusterRoles(t *testing.T) {
+	assert := assert.New(t)
+
+	modelBindingPair := testutil.GetModelBindingPair()
+	clusterConnections := testutil.GetManagedClusterConnections()
+	clusterConnection := clusterConnections["cluster1"]
+
+	CreateClusterRoles(modelBindingPair, clusterConnections)
+
+	list, err := clusterConnection.KubeClient.RbacV1().ClusterRoles().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("can't list cluster roles: %v", err)
+	}
+	assert.Equal(1, len(list.Items))
+	cr := list.Items[0]
+	assert.Equal("verrazzano-system", cr.Name)
+	assertRules(t, &cr, getRules())
+}
+
+func TestCreateClusterRolesVmiSystem(t *testing.T) {
+	assert := assert.New(t)
+
+	modelBindingPair := testutil.GetModelBindingPair()
+	clusterConnections := testutil.GetManagedClusterConnections()
+	clusterConnection := clusterConnections["cluster1"]
+
+	modelBindingPair.Binding.Name = constants.VmiSystemBindingName
+
+	CreateClusterRoles(modelBindingPair, clusterConnections)
+
+	list, err := clusterConnection.KubeClient.RbacV1().ClusterRoles().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("can't list cluster roles: %v", err)
+	}
+	expectedRoles := monitoring.GetSystemClusterRoles("cluster1")
+	assert.Equal(len(expectedRoles), len(list.Items))
+
+	for _, expected := range expectedRoles {
+		cr, err := clusterConnection.KubeClient.RbacV1().ClusterRoles().Get(context.TODO(), expected.Name, metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("can't get cluster role %s: %v", expected.Name, err)
+		}
+		assertRules(t, cr, expected.Rules)
+	}
+}
+
+func TestCreateClusterRolesUpdate(t *testing.T) {
+	assert := assert.New(t)
+
+	modelBindingPair := testutil.GetModelBindingPair()
+	clusterConnections := testutil.GetManagedClusterConnections()
+	clusterConnection := clusterConnections["cluster1"]
+
+	cr := v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   util.GetServiceAccountNameForSystem(),
+			Labels: util.GetManagedLabelsNoBinding("cluster1"),
+		},
+	}
+	_, err := clusterConnection.KubeClient.RbacV1().ClusterRoles().Create(context.TODO(), &cr, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("can't create cluster role: %v", err)
+	}
+
+	CreateClusterRoles(modelBindingPair, clusterConnections)
+
+	list, err := clusterConnection.KubeClient.RbacV1().ClusterRoles().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("can't list cluster roles: %v", err)
+	}
+	assert.Equal(1, len(list.Items))
+	cr = list.Items[0]
```

Comment:
apart from these few lines - this test seems to be the same as TestCreateClusterRoles

Reply:
This test sets the binding name to 'system' which causes a completely different code path in CreateClusterRoles.  It creates the system cluster roles in that case.  I've updated the comments to make that clearer, I hope.